### PR TITLE
Refactor styling to .p-code-numbered

### LINKED ISSUE #586: [Proposal] Grid-list and image-grid should be renamed
# Pattern amendment

## Is this a proposal or an amendment?

- Amendment

`.p-grid-list` / `.p-image-grid`

## Context

in my opinion, the naming of these patterns is misleading as neither of them have anything to do with the actual grid which could easily lead to confusion given the underpin tenet of all CSS framework tends to be a grid.

## Proposed action

Rename both of these patterns so they don't reference 'grid'

### LINKED ISSUE #856: Update code numbered pattern
## Is this a proposal or an amendment?

- Amendment

If amendment, what is the name of the pattern to be amended?
Code numbered

## Visual

Current build with right-aligned line numbers:
code numbered vanilla framework examples2

Proposal build with background-color behind the line numbers:
code numbered vanilla framework examples

## Context
A recent update to the numbered code block required the line numbers to right-align as to prevent the numbers with double digits pushing out single digit number lines. This requires the number area to be a set width to allow room for the numbers to grow to the left.

This styling makes it clear the numbers are separate from the code content. Giving a clear indication of the length of growth too.

The current line number growth with the attached design is 9999.

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Refactor styling to .p-code-numbered

## QA

- Pull code
- Run `gulp jekyll`
- Check [example](http://127.0.0.1:4000/vanilla-framework/examples/patterns/code-numbered/) against the [spec](https://github.com/ubuntudesign/vanilla-design/blob/master/Code/code.md)

## Details

Fixes #856

### CODE REVIEW

**Comment by yaili** [comment]
Are you sure it's #586?

**Comment by barrymcgee** [comment]
Oops, should be #856

File: scss/_patterns_code-numbered.scss [hunk 1]
```diff
@@ -4,18 +4,44 @@
...
-      counter-increment: line-numbering;
-      display: inline-block;
-      padding-right: $sp-medium; // space after numbers
-      pointer-events: none;
-      text-align: right;
-      user-select: none;
-      width: $sp-xxx-large;
+    code {
+      padding: 0 0 0;
+      position: relative;
```
Comment (anthonydillon) [inline scss/_patterns_code-numbered.scss]: It doesn't look like these styles change anything and we should not target element in patterns.

**Comment by anthonydillon** [comment]
When the line wraps it should only wrap within the grey area and the white number area should be the full height. Also on small screens there is a small grey line between the numbers.

**Comment by barrymcgee** [comment]
Ready for re-review 👀

File: scss/_patterns_code-numbered.scss [hunk 2]
```diff
@@ -1,21 +1,56 @@
...
+    &::before {
+      background-color: $color-x-light;
+      width: $sidebar-width;
+    }
+
+    .code-line {
+      background: $color-light;
+      display: block;
+      margin: -#{$sp-large} 0 0 0;
+      padding: $sp-x-small 0 $sp-x-small 5.5rem;
```
Comment (anthonydillon) [inline scss/_patterns_code-numbered.scss]: Shouldn't the left padding here be relative to `$sidebar-width + 1rem`

File: scss/_patterns_code-numbered.scss [hunk 3]
```diff
@@ -1,21 +1,56 @@
...
+
+  $sidebar-width: 4.5rem;
+
   .p-code-numbered {
-    background: $color-light;
+    background: $color-x-light;
     color: $color-dark;
     counter-reset: line-numbering;
-    padding: $sp-medium;
+    padding: 1rem 0 0;
```
Comment (anthonydillon) [inline scss/_patterns_code-numbered.scss]: Can this `1rem` be a spacing value? `$sp-???`

**Comment by anthonydillon** [comment]
The line spacing between lines of code is too large:

**Comment by anthonydillon** [comment]
The text gets close to the right hand side. I think it would be better to have equal padding.

**Comment by anthonydillon** [comment]
There are small white lines between code on small screens:

**Pull request merged**